### PR TITLE
GSW-1838 refactor: protocol fee withdrawal

### PR DESCRIPTION
**Pull request by onlyhyde**
- This commit use origin admin address (g1lmvrrrr4er2us84h2732sru76c9zl2nvknha8c)
- Some of token_register.gno code are modified (remove locked variable)
- Re-entry attacks are not possible in GNO, so the locked flag is removed. Also, due to changes in the grc20 spec, a panic may occur in the grc20 contract code, in which case the locked flag is not changed to false and terminated, locking the state value.

### CODE REVIEW

File: pool/_helper_test.gno [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+func MintPosition(t *testing.T,
+	token0 string,
+	token1 string,
+	fee uint32,
+	tickLower int32,
+	tickUpper int32,
+	_amount0Desired string, // *u256.Uint
+	_amount1Desired string, // *u256.Uint
+	_amount0Min string, // *u256.Uint
+	_amount1Min string, // *u256.Uint
```
Comment (notJoon) [inline pool/_helper_test.gno:303]: need to remove prefix

```suggestion
	amount0Desired string, // *u256.Uint
	amount1Desired string, // *u256.Uint
	amount0Min string, // *u256.Uint
	amount1Min string, // *u256.Uint
```

File: pool/_helper_test.gno [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+	return pn.Mint(
+		token0,
+		token1,
+		fee,
+		tickLower,
+		tickUpper,
+		_amount0Desired,
+		_amount1Desired,
+		_amount0Min,
+		_amount1Min,
```
Comment (notJoon) [inline pool/_helper_test.gno:320]:
```suggestion
		amount0Desired,
		amount1Desired,
		amount0Min,
		amount1Min,
```

**Review by notJoon — CHANGES_REQUESTED** [review]
left some minor fix. LGTM 👍

File: pool/_helper_test.gno [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+	TokenFaucet(t, gnsPath, alice)
+	TokenApprove(t, wugnotPath, alice, pool, uint64(1000))
+	TokenApprove(t, gnsPath, alice, pool, uint64(1000))
+	MintPosition(t,
+		wugnotPath,
+		gnsPath,
+		fee3000,
+		int32(1020),
+		int32(5040),
+		("1000"),
```
Comment (r3v4s) [inline pool/_helper_test.gno:193]:
```suggestion
		"1000",
```

I don't think we need bracket

File: pool/_helper_test.gno [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+	}
+
+	return uint64(coins.AmountOf(ugnotDenom))
+}
+
+func ugnotMint(t *testing.T, addr std.Address, denom string, amount int64) {
+	t.Helper()
+	banker := std.GetBanker(std.BankerTypeRealmIssue)
+	banker.IssueCoin(addr, denom, amount)
+	std.TestIssueCoins(addr, std.Coins{{denom, int64(amount)}})
```
Comment (r3v4s) [inline pool/_helper_test.gno:438]: seems like minting twice

File: pool/_helper_test.gno [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+		ugnotMint(t, faucetAddress, newCoins[0].Denom, newCoins[0].Amount)
+		std.TestSetOrigSend(newCoins, nil)
+	}
+	ugnotTransfer(t, faucetAddress, to, amount)
+}
+
+func ugnotDeposit(t *testing.T, addr std.Address, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(addr))
+	wugnotAddr := std.DerivePkgAddr(consts.WRAPPED_WUGNOT)
```
Comment (r3v4s) [inline pool/_helper_test.gno:463]: this addr is defined in r/gnoswap/v1/const

**Review by r3v4s — APPROVED** [review]

**Comment by r3v4s** [comment]
@onlyhyde I think admin address needs to be updated.

**Pull request merged**